evalml/automl/automl_search.py

```diff
@@ -205,22 +205,21 @@ def __init__(self,
         additional_objectives = [self._validate_objective(obj) for obj in additional_objectives]
         self.additional_objectives = additional_objectives
 
-        if max_time is None or isinstance(max_time, (int, float)):
-            self.max_time = max_time
-        elif isinstance(max_time, str):
-            self.max_time = convert_to_seconds(max_time)
-        else:
-            raise TypeError("max_time must be a float, int, or string. Received a {}.".format(type(max_time)))
-
+        if not isinstance(max_time, (int, float, str, type(None))):
+            raise TypeError(f"Parameter max_time must be a float, int, string or None. Received {str(max_time)}.")
```

Comment:
Would it be more useful to include the type of `max_time` passed, ie `type(max_time)`, in the error message?

```python
TypeError(f"Parameter max_time must be a float, int, string or None. Received {type(max_time)} with value {str(max_time)}.")
```
seems to be more helpful as a `TypeError` message.

Reply:
Agreed!